### PR TITLE
Finalize WINDOWEVENT API, minor fixes

### PR DESCRIPTION
**Pull request by ankith26**
Finalizing the new API, getting rid of the old one. Some more fixes as well. I’m currently testing event stuff a lot, just to make sure I’m not breaking stuff

### CODE REVIEW

**Comment by ankith26** [comment]
This PR can close #2214

File: src_c/event.c [hunk 1]
```diff
@@ -598,12 +594,22 @@ _pg_pgevent_deproxify(Uint32 type)
 #if IS_SDLv2
 static SDL_Event *_pg_last_keydown_event = NULL;
 
+static int
+_pg_translate_windowevent(void *_, SDL_Event *event)
+{
+    if (event->type == SDL_WINDOWEVENT) {
+        event->type = PGE_WINDOWSHOWN + event->window.event - 1;
+        return SDL_EventState(_pg_pgevent_proxify(event->type), SDL_QUERY);
```
Comment (illume) [inline src_c/event.c:602]: What's going on here?

File: src_c/event.c [hunk 2]
```diff
@@ -1751,20 +1753,75 @@ get_grab(PyObject *self)
...
+    while (1) {
+        _pg_event_pump(1); /* Use our custom pump here */
+        switch (PG_PEEP_EVENT_ALL(event, 1, SDL_GETEVENT)) {
+            case -1:
+                return 0; /* Because this never happens, SDL does it too*/
+            case 1:
+                return 1;
+
+            default:
+                if (timeout >= 0 && SDL_GetTicks() >= finish) {
```
Comment (illume) [inline src_c/event.c:1791]: I went through this function in detail and it looks ok.

File: src_c/event.c [hunk 3]
```diff
@@ -1751,20 +1753,75 @@ get_grab(PyObject *self)
...
+     * might break. So after every event pump, we translate events from
+     * here */
+    SDL_FilterEvents(_pg_translate_windowevent, NULL);
+#endif
+}
+
+static int
+_pg_event_wait(SDL_Event *event, int timeout)
+{
+    /* Custom re-implementation of SDL_WaitEventTimeout, doing this has
```
Comment (illume) [inline src_c/event.c:1774]: Can we write here what the advantages are? I guess it mainly uses our pump/peek implementations?

File: test/event_test.py [hunk 4]
```diff
@@ -559,23 +558,15 @@ def test_wait(self):
...
 
         # Test case with timeout and an event in the queue.
-        if SDL1:
-            with self.assertRaises(TypeError):
-                pygame.event.wait(250)
-        else:
-            event = pygame.event.Event(EVENT_TYPES[0], **EVENT_TEST_PARAMS[EVENT_TYPES[0]])
-            pygame.event.post(event)
-            wait_event = pygame.event.wait(250)
+        event = pygame.event.Event(EVENT_TYPES[0], **EVENT_TEST_PARAMS[EVENT_TYPES[0]])
```
Comment (illume) [inline test/event_test.py:565]: Nice the wait works with SDL1 :)

File: src_c/event.c (hunk 1 above)
Comment (ankith26) [reply src_c/event.c:602]: That’s where we do SDL_WINDOWEVENT translation into respective events. I figured out the old approach to do that from the event filter has its disadvantages, so we do it here now

File: src_c/event.c (hunk 2 above)
Comment (ankith26) [reply src_c/event.c:1791]: That was mostly copied from SDL, just put a bit of twists into it :)

File: src_c/event.c (hunk 3 above)
Comment (ankith26) [reply src_c/event.c:1774]: Yes. We have our own pump function, that we should be using, going forward. This is to trigger the SDL_WINDOWEVENT translation

File: test/event_test.py (hunk 4 above)
Comment (ankith26) [reply test/event_test.py:565]: That’s another advantage of doing the custom wait implementation :)

**Review by illume — APPROVED** [review]
Thanks :)  I did a bit of manual testing, and it LGTM.

File: src_c/event.c (hunk 2 above)
Comment (nimagoli) [reply src_c/event.c:1791]: عالیست

**Pull request merged**